File: 43.multiply-strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var multiply = function(num1, num2) {
    let m = num1.length;
    let n = num2.length;
    let ans = new Array(m + n).fill(0);
    num1 = num1.split("").reverse();
    num2 = num2.split("").reverse();
    for(let i = 0; i < m; i++) {
        let carry = 0;
        for(let j = 0; j < n; j++) {
            let num = (ans[i + j]) + (num1[i] - '0') * (num2[j] - '0') + carry;
            ans[i + j] = num % 10;
            carry = Math.floor(num / 10);
        }
        ans[i + n] += carry;
    }
    ans.reverse();
    while (ans.length > 1 && ans[0] == 0) {
        ans.shift();
    }
    return ans.join("");
};

File: 43.multiply-strings.py
class Solution:
    def multiply(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        m = len(num1)
        n = len(num2)
        ans = [0] * (m + n)
        num1 = [int(x) for x in reversed(num1)]
        num2 = [int(x) for x in reversed(num2)]
        for i in range(m):
            carry = 0
            for j in range(n):
                num = ans[i + j] + num1[i] * num2[j] + carry
                ans[i + j] = num % 10
                carry = num // 10
            ans[i + n] += carry
        ans = "".join(str(x) for x in reversed(ans)).lstrip('0')
        return ans or '0'